### ISSUE TITLE
Improving Hack tooling API and PHPStorm IDE Support 

### ISSUE THREAD

**Issue by anik786** [facebook/hhvm#6940]
According to http://blog.jetbrains.com/phpstorm/2015/06/hack-language-support-in-phpstorm-postponed/ Jetbrains are refusing to extend their IDE to support Hack due to the following: "We’re unable to deliver the level of functionality & quality expected from an IDE by JetBrains, with the **currently available Hack tooling APIs**." 

The article continues by mentioning numerous problems with the current Hack tooling APIs and concluding "Given the above, we’ve decided to postpone Hack language support in PhpStorm until either **Hack language seems greater adoption** or the **options for integrating with the original tooling change dramatically**."

This is a shame, as PHPStorm is without doubt the best PHP IDE there is and Hack Lang is a HUGE improvement over PHP. IDE support has also been hugely requested as mentioned in the article as well as all over Stackoverflow.

Bearing in mind the article was written almost a year ago; my question is: 
When will Hack Lang start to better support other IDE's?

This is vital in enabling Hack Lang gaining greater adoption, and perhaps the only thing stopping many developers using Hack.

**Comment by javacty** [comment]
Sincerely hope that you cooperate，PHPStorm is really good tool for developing php.

**Comment by fredemmott** [comment]
- that post was a surprise to us and we did not feel it was representative of our efforts
 - we believe Nuclide shows that sufficient hooks are available
 - if contributors to another IDE ask for specific features, I'm not working on Hack/HHVM any more, but I'll try to get them prioritized
 - PHPStorm is the clear leader in PHP IDEs - but *we* can't contribute this feature to it. Given the size of the HHVM OSS team and relative popularity of other IDEs, directly contributing to other IDEs is not a good use of our time.

Closing as we can't act on this issue - more specific requests needed.
